core-test/src/test/scala/org/bitcoins/core/crypto/SchnorrTest.scala

```diff
@@ -0,0 +1,166 @@
+package org.bitcoins.core.crypto
+
+import org.bitcoin.NativeSecp256k1
+import org.bitcoins.core.util.{CryptoUtil, NumberUtil}
+import org.bitcoins.testkit.core.gen.{CryptoGenerators, NumberGenerator}
+import org.bitcoins.testkit.util.BitcoinSUnitTest
+import scodec.bits.ByteVector
+
+class SchnorrTest extends BitcoinSUnitTest {
+  implicit override val generatorDrivenConfig: PropertyCheckConfiguration =
+    generatorDrivenConfigNewCode
+
+  behavior of "NativeSecp256k1"
+
+  it must "act like a normal schnorrSign if the normal nonce is specified" in {
+    forAll(CryptoGenerators.nonZeroPrivKey, NumberGenerator.bytevector(32)) {
+      case (privKey, message) =>
+        val nonce = SchnorrNonce.fromBipSchnorr(privKey, message)
+
+        assert(nonce.bytes.toArray.exists(_ != 0.toByte))
+
+        val sigWithNonce =
+          NativeSecp256k1.schnorrSignWithNonce(message.toArray,
+                                               privKey.bytes.toArray,
+                                               nonce.bytes.toArray)
+
+        assert(
+          SchnorrDigitalSignature(ByteVector(sigWithNonce)) ==
+            Schnorr.signWithNonce(message, privKey, nonce)
+        )
+
+        val sig =
+          NativeSecp256k1.schnorrSign(message.toArray, privKey.bytes.toArray)
+
+        assert(
```

Comment:
//make sure secp256k1 deterministically signs when just passing in a message and a privKey